src/main/java/de/tum/in/www1/artemis/service/connectors/localvc/LocalVCServletService.java

```diff
@@ -238,13 +243,37 @@ public void authenticateAndAuthorizeGitRequest(HttpServletRequest request, Repos
             throw new LocalVCForbiddenException();
         }
 
-        authorizeUser(repositoryTypeOrUserName, user, exercise, repositoryAction, localVCRepositoryUri.isPracticeRepository());
+        var authenticationMechanism = resolveAuthenticationMechanism(authorizationHeader, user);
+        var ipAddress = request.getRemoteAddr();
+        authorizeUser(repositoryTypeOrUserName, user, exercise, repositoryAction, authenticationMechanism, ipAddress, localVCRepositoryUri);
 
         request.setAttribute("user", user);
 
         log.debug("Authorizing user {} for repository {} took {}", user.getLogin(), localVCRepositoryUri, TimeLogUtil.formatDurationFrom(timeNanoStart));
     }
 
+    /**
+     * Resolves the user's authentication mechanism for the repository
+     *
+     * @param authorizationHeader the request's authorizationHeader, containing the token or password
+     * @param user                the user
+     * @return the authentication type
+     * @throws LocalVCAuthException if extracting the token or password from the authorizationHeader fails
+     */
+    private AuthenticationMechanism resolveAuthenticationMechanism(String authorizationHeader, User user) throws LocalVCAuthException {
+        UsernameAndPassword usernameAndPassword = extractUsernameAndPassword(authorizationHeader);
+
+        String password = usernameAndPassword.password();
+        if (!password.startsWith("vcpat")) {
```

Comment:
I think in the "Generate Token" PR #9182, you use a constant instead of the direct prefix `"vcpat"` for better readability. Please also do so here 👍